client/state/purchases/test/reducer.js

```diff
@@ -33,27 +33,29 @@ describe( 'reducer', () => {
 	} );
 
 	test( 'should return an object with the list of purchases when fetching completed', () => {
+		const siteId2 = '2828828282';
 		let state = reducer( undefined, {
 			type: PURCHASES_USER_FETCH_COMPLETED,
 			purchases: [
-				{ ID: '1', blog_id: siteId, user_id: userId },
+				{ ID: '1', blog_id: siteId2, user_id: userId },
 				{ ID: '2', blog_id: siteId, user_id: userId },
 			],
 		} );
 
 		state = reducer( state, {
 			type: PURCHASES_SITE_FETCH_COMPLETED,
 			purchases: [
-				{ ID: '2', blog_id: siteId, user_id: userId },
+				{ ID: '2', blog_id: siteId, user_id: userId, other: true },
 				{ ID: '3', blog_id: siteId, user_id: userId },
 			],
+			siteId: Number( siteId ),
 		} );
 
 		expect( state ).toEqual( {
 			data: [
-				{ ID: '2', blog_id: siteId, user_id: userId },
+				{ ID: '1', blog_id: siteId2, user_id: userId },
```

Comment:
id 1 should remain in the same place unchanged because it is for a different site than the recent site request.

id 2 should be updated in-place because it was returned by the site request and already existed.

id 3 should be added because it was added by the site request.